tests/unit/lms/views/api/lti_test.py

```diff
@@ -12,14 +12,15 @@
 
 class TestRecordCanvasSpeedgraderSubmission:
     def test_it_passes_correct_params_to_read_current_score(
-        self, pyramid_request, lti_outcomes_client
+        self, ai_getter, pyramid_request, lti_outcomes_client
     ):
         LTIOutcomesViews(pyramid_request).record_canvas_speedgrader_submission()
 
+        ai_getter.shared_secret.assert_called_once_with("TEST_OAUTH_CONSUMER_KEY")
         lti_outcomes_client.read_result.assert_called_once_with(
             LTIOutcomesRequestParams(
                 consumer_key="TEST_OAUTH_CONSUMER_KEY",
-                shared_secret="oauth-secret",
+                shared_secret=ai_getter.shared_secret.return_value,
```

Comment:
There's no need to set `ai_getter.shared_secret.return_value` to `"oauth-secret"`. Mock methods already have return values (each mock method returns another unique mock object) and they can just be accessed as `.return_value` (and nothing in these tests cares that the return value is actually a string)

Reply:
This is quite nice. I've not done it before, but good to know